### DISCUSSION TITLE
Setting date time [Oct 26]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#4267] [Discord Threads]
> 🙋‍♂️ **vikiiingur**  
#### If you follow the order year+month+day then the date changes correctly.  
![image0](https://cdn.discordapp.com/attachments/661905455894888493/1031953327405158451/unknown.png)

**Comment by o1lab** [comment]
> 💁 **o1lab**  
#### https://github.com/nocodb/nocodb/issues/4129  
#  
> 🙋‍♂️ **wkw**  
#### I don't think this is a bug.    
####    
#### First the date type is simply the format of how we display in the cell, not the order how you input in the datepicker. Second, I've tested the order in the date picker library and I get the same behaviour, which is once you change the month or the year, you still need to select the day. Otherwise, the value won't be updated.   
####    
#### I guess the reason behind is    
####    
#### If I choose 29/02/2020 (leap day), then i update the month, let's say May. if the date changes automatically, then it would be 29/05/2020 which would be an invalid date. Besides, for UX, i think most people choose from year to day.  
#  
> 🙋‍♂️ **vikiiingur**  
#### I see, thanks for the update, maybe a visual feedback would be useful to remind that once you change a part of the date, the whole date needs to be re-added. So far it is not visible, and when I clicked away, I just noticed by mistake that the date did not change and rolled back to the previous value.  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1031953327862321224)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
